google/cloud/firestore_v1/order.py

```diff
@@ -165,6 +176,19 @@ def compare_arrays(left, right) -> int:
 
         return Order._compare_to(len(l_values), len(r_values))
 
+    @staticmethod
+    def compare_vectors(left, right) -> int:
```

Comment:
I see what you mean, but I don't think using `Order.compare_arrays` is necessary. In fact, since we know each element of the array is a float, using `Order.compare_arrays` feels a bit wasteful (we don't need to recurse into `Order.compare` for each element, or the array length comparison logic)

If I'm understanding right, we should get the same result with something like this:
```
def __lt__(other):
  length = len(self.value)
  if length != len(other.value):
    return length < len(other.value)
  
  for i in range(length):
    if self.value[i] != other.value[i]:
       return self.value[i] < other.value[i]
   return False
```

---

But actually, looking agin, maybe we don't have access to the actual Vector class here, only the proto representation?

Reply:
Yes the TypeOrder does not perform type conversion (no dependency on Vector or any type) but purely compare the proto format.